src/lib/upgradePackageDefinitions.ts

```diff
@@ -49,17 +120,55 @@ export async function upgradePackageDefinitions(
 
   if (options.peer && Object.keys(filteredLatestDependencies).length > 0) {
     const upgradedPeerDependencies = await getPeerDependenciesFromRegistry(filteredLatestDependencies, options)
-    const peerDependencies = { ...options.peerDependencies, ...upgradedPeerDependencies }
-    if (!dequal(options.peerDependencies, peerDependencies)) {
-      const [newUpgradedDependencies, newLatestVersions, newPeerDependencies] = await upgradePackageDefinitions(
-        { ...currentDependencies, ...filteredUpgradedDependencies },
-        { ...options, peerDependencies, loglevel: 'silent' },
+    let checkPeerViolationResult = checkIfInPeerViolation(
+      currentDependencies,
+      filteredUpgradedDependencies,
+      upgradedPeerDependencies,
+    )
+    // Try another run to see if we can upgrade some more
+    let rerunResult = await rerunUpgradeIfChangedPeers(
+      currentDependencies,
+      filteredUpgradedDependencies,
+      upgradedPeerDependencies,
+      latestVersionResults,
+      options,
+    )
+    if (!rerunResult) {
```

Comment:
There is a lot of duplication between the second run (step 2) and the while loop (step 3). Can you combine this into a single while or do-while loop? You may be able to prepopulate `checkPeerViolationResult` to combine the logic.

Reply:
was hard :-) but done